Python/Module2_EssentialsOfPython/Basic_Objects.md

```diff
@@ -721,11 +721,77 @@ True
 
 ```
 
+### Formatting strings
+Python provides multiple syntaxes for formatting strings; these permit us to do things like programmatically inject the values of variables into strings, align fields using whitespace, and control the number of decimal places with which numbers are displayed in a string. This section is designed to simply expose the reader to the different varieties of string-formatting. 
+
+[pyformat.info](https://pyformat.info) is the best resource to consult to see an exhaustive (but still intuitive) treatment of string-formatting in Python. You can also refer to the official documentation [here](https://docs.python.org/3/library/string.html#format-examples).
+
+In Python 3, you can leverage the `format` method towards this end:
+
+```python
+# using `format` to replace placeholders with values
+>>> name = "Bruce"
+>>> age = 80
+>>> "{name} is {age} years old".format(name="Bruce", age=80)
+'Bruce is 80 years old'
+
+# padding a string with 4 leading spaces
+>>> "{item:>4}".format(item="stew")
+'   stew'
+```
+Note that you may encounter the use of the cryptic `%` operator to format strings to the same effect:
+
+```python
+# using `%` to  format strings (avoid this)
+>>> name = "Selina"
+>>> "My name is %s" % name
+'My name is Selina'
+```
+this is a relic of Python 2; it is recommend that you avoid this formatting syntax.
+
+If you are using Python 3.6 or beyond, then you have the luxury of being able to use f-strings, which provide a supremely convenient means for formatting strings. Here is an example of an f-string in action:
+
+```python
+# an example of an 'f-string'
+>>> batman = 12
+>>> catwoman = 10
+>>> f"Batman has {batman} apples. Catwoman has {catwoman} apples. Together, they have {batman + catwoman} apples"
+'Batman has 12 apples. Catwoman has 10 apples. Together, they have 22 apples'
+```
+
+See that an f-string has a special syntax; an f-string is denoted by preceding the opening quotation mark with the lowercase f character:
+```python
+# this is a typical empty string
+>>> ""
+''
+
+# this is an empty f-string
+>>> f""
+''
+```
+
```

Comment:
This is your call but I don't think I'd put the brackets inside parens.

"within curly brackets (`{}`) will be"

"within curly brackets `{}` will be"

....I kind of don't like either. Maybe

"within curly brackets (e.g. `{x - 3}`) will be"

or something.

Reply:
Good call